astroid/bases.py

```diff
@@ -120,7 +122,11 @@ def infer(self, context=None):
         yield self
 
 
-def _infer_stmts(stmts, context, frame=None):
+def _infer_stmts(
+    stmts: list[InferenceResult],
```

Comment:
Even if technically correct, this doesn't look right. Why should we call a function which takes `InferenceResults` as inputs only to also return `InferenceResults`. The Union is probably better. One other thing, maybe `Sequence` instead of `list`?

Reply:
So `InferenceResult` as input and the `Union` of `InferenceResult` as return value?

Updated to `Sequence`.

Reply:
> So `InferenceResult` as input and the `Union` of `InferenceResult` as return value?

Switched. `stmts: Union[...]` and `-> InferenceResult`. Result and return annotation does make sense IMO.